File: python/2/solution_test.py
from solution import solution
import random

def makeArray(size):
    return [random.randint(-1000,1000) for x in range(size)]

inputs = [
    ([3, 8, 9, 7, 6], 3),
    ([0, 0, 0], 1),
    ([1, 2, 3, 4], 4),
    ([-100, -50, 123, 0, 393, 123, 32, -1], 50),
    (makeArray(100), 100),
    ([], 3)
]

outputs = [
    [9, 7, 6, 3, 8],
    [0, 0, 0],
    [1, 2, 3, 4]
]

def validate(result, expected):
    for i, val in enumerate(result):
        assert val == expected[i]


def test_solution():
    for i, val in enumerate(inputs):
        result = solution(*val)
        
        if i < len(outputs):
            validate(result, outputs[i])
        else:
            print("Returned {}".format(result))

File: go/3/tape-equilibrium/solution_test.go
package solution

import "testing"

var (
	inputs = [][]int{
		[]int{
			3, 1, 2, 4, 3,
		},
		[]int{
			4, 3, 5, 2, 8,
		},
		[]int{},
		[]int{
			1, 200,
		},
	}

	outputs = []int{
		1, 2, 0, 199,
	}
)

func TestSolution(t *testing.T) {
	for i, input := range inputs {
		expected := outputs[i]
		actual := Solution(input)
		if actual != expected {
			t.Fatal("expected ", expected, " actual ", actual)
		}
	}
}
